sdks/node/src/weaveClient.ts

```diff
@@ -295,14 +296,49 @@ export class WeaveClient {
     }
 
     const t = val?._type;
+
+    if (t == 'StringPrompt') {
+      const {StringPrompt} = await import('./prompt');
+
+      let obj = new StringPrompt({
+        id: dataObj.id,
+        description: val.description,
+        content: val.content,
+      });
+
+      obj.__savedRef = ref;
+
+      return obj;
+    }
+
+    if (t == 'MessagesPrompt') {
+      const {MessagesPrompt} = await import('./prompt');
+
+      let obj = new MessagesPrompt({
+        id: dataObj.id,
+        description: val.description,
+        messages: val.messages,
+      });
+
+      obj.__savedRef = ref;
+
+      return obj;
+    }
+
     if (t == 'Dataset') {
-      const {_baseParameters, rows} = val;
+      // Avoid circular dependency
+      const {Dataset} = await import('./dataset');
+
+      const {description, rows} = val;
+
       let obj = new Dataset({
-        id: _baseParameters.id,
-        description: _baseParameters.description,
+        id: dataObj.id,
+        description: description,
```

Comment:
Serialization/deserialization should not rely on `_baseParameters`. This property is not available on objects serialized by the Python SDK.